### PR TITLE
Fix for TC test_container_checker.py issue https://github.com/sonic-net/sonic-mgmt/issues/10475 to check if the container is in running state before stopping it 

### PR DESCRIPTION
**Pull request by vikshaw-Nokia**
### Description of PR

Summary:

Refer: Issue https://github.com/sonic-net/sonic-mgmt/issues/10475
In the resolution process, we verify the status of the container under test to ensure it is running. If the container is found to be in exited stated, we initiate a config_reload to guarantee its running state before proceeding with the test case. To maintain accurate logging post config_reload, an additional 10-second delay is introduced for the log analyzer, compensating for the reset of the monit rc timer.

In certain scenarios, the auto-restart feature may inadvertently get enabled due to other test cases, which contradicts the prerequisite of having it disabled for the current test case. Consequently, a corrective measure has been implemented to ensure that auto-restart is appropriately disabled as part of the fix.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205
- [ ] 202305
- [ ] 202311

#### What is the motivation for this PR?
Issue: https://github.com/sonic-net/sonic-mgmt/issues/10475
#### How did you do it?
If the container in test is not running -> Perform config_reload to ensure its running state
Sleep time is increased for 10s in case of config_reload for Log_Analyzer
If the auto_restart feature is enabled -> Perform disable_container_autorestart to ensure the auto restart is disabled

#### How did you verify/test it?
Ran the test cases against a multi-asic line card in a T2 chassis.

### Documentation
image

### CODE REVIEW

File: tests/container_checker/test_container_checker.py [hunk 1]
```diff
@@ -203,18 +204,26 @@ def test_container_checker(duthosts, enum_rand_one_per_hwsku_hostname,
...
+    feature_autorestart_states = duthost.get_container_autorestart_states()
+    if feature_autorestart_states.get(service_name) == 'enabled':
+        disable_container_autorestart(duthost)
+        time.sleep(30)
+    if not is_container_running(duthost, container_name):
+        logger.info("Container '{}' is not running ...".format(container_name))
+        logger.info("Reload config on DuT as Container is not up '{}' ...".format(duthost.hostname))
+        config_reload(duthost, safe_reload=True)
+        time.sleep(300)
+        sleep_time = 80
```
Comment (SuvarnaMeenakshi) [inline tests/container_checker/test_container_checker.py:226]: why do we need to modify the sleep_time to 80 if the container is not running, given that there is alrady sleep(300) ?

File: tests/container_checker/test_container_checker.py [hunk 2]
```diff
@@ -203,18 +204,26 @@ def test_container_checker(duthosts, enum_rand_one_per_hwsku_hostname,
...
                    "Container '{}' is skipped for testing.".format(container_name))
-
+    feature_autorestart_states = duthost.get_container_autorestart_states()
+    if feature_autorestart_states.get(service_name) == 'enabled':
+        disable_container_autorestart(duthost)
+        time.sleep(30)
+    if not is_container_running(duthost, container_name):
+        logger.info("Container '{}' is not running ...".format(container_name))
+        logger.info("Reload config on DuT as Container is not up '{}' ...".format(duthost.hostname))
+        config_reload(duthost, safe_reload=True)
```
Comment (SuvarnaMeenakshi) [inline tests/container_checker/test_container_checker.py:224]: if container autorestart is disabled (line 219) and that is not saved in the config, then doing a config reload will set it back to what was in the saved config, is that intentional?

File: tests/container_checker/test_container_checker.py (hunk 1 above)
Comment (vikshaw-Nokia) [reply tests/container_checker/test_container_checker.py:226]: @SuvarnaMeenakshi 
 The Particular Sleep time is set for the log Analyzer, which is 70s in case the container was in start state, and 80s in case of config reload. A Increase of 10s is added to ensure logging in syslogs for the latter case. 
image

File: tests/container_checker/test_container_checker.py (hunk 2 above)
Comment (vikshaw-Nokia) [reply tests/container_checker/test_container_checker.py:224]: @SuvarnaMeenakshi 
Config reload will not enable the container auto restart, only a load minigraph will as the config is saved in the function  disable_container_autorestart

image

File: tests/container_checker/test_container_checker.py (hunk 1 above)
Comment (vikshaw-Nokia) [reply tests/container_checker/test_container_checker.py:226]: @SuvarnaMeenakshi , After the config reload the timer of container checker get reset, and issue is observed , as per the /etc/monit/conf.d/sonic-host the image attached for reference, A 50 Second delay is observed between the checks when no issue is observed, which and the daemon is set for 10s with start delay of 10s, So in order to escape the corner case an extra wait of 10s is added.
image

Comment (SuvarnaMeenakshi) [reply tests/container_checker/test_container_checker.py:226]: ok thank you

**Review by SuvarnaMeenakshi — APPROVED** [review]

**Review by judyjoseph — APPROVED** [review]

**Comment by vikshaw-Nokia** [comment]
PR for 2205 is created https://github.com/sonic-net/sonic-mgmt/pull/12155
